Sources/Core/Sources/Resources/Texture/Texture.swift

```diff
@@ -179,6 +179,21 @@ public struct Texture {
     bytes[index + 2] = pixel.r
     bytes[index + 3] = pixel.a
   }
+    
+  /// Sets the alpha components of the texutre to a specified value. Alpha value is bound to a range (0...255 inclusive).
+  public mutating func setAlpha(_ alpha: UInt8) {
+    for x in 0..<width {
+      for y in 0..<height {
+        let pixel = getPixel(atX: x, y: y)
+        let newPixel = Pixel(
+          r: UInt8(pixel.r),
+          g: UInt8(pixel.g),
+          b: UInt8(pixel.b),
+          a: UInt8(min(max(alpha, 0), 255)))
```

Comment:
Calculate the bounded value (`min(max(alpha, 0), 255)`) at the start of the function just to avoid the unnecessary calculations. Probably not completely necessary but it might matter if the user has a high resolution texture pack? Just good to be safe.